ext/zip/tests/bug76524.phpt

```diff
@@ -0,0 +1,25 @@
+--TEST--
+ZipArchive Bug #76524 (memory leak with ZipArchive::OVERWRITE flag and empty archive)
+--SKIPIF--
+<?php
+/* $Id$ */
+if(!extension_loaded('zip')) die('skip');
+?>
+--FILE--
+<?php
+
+$filename = __DIR__ . '/nonexistent.zip';
+
+// The error is not reproduced when file already exist:
+if (!file_exists($filename)) {
```

Comment:
Why not just ensure the filename doesn't exist?

Reply:
Btw., how would it behave if something was added to that non existent archive? Perhaps this case deserves a separate test.

Thanks.

Reply:
> Why not just ensure the filename doesn't exist?

Of course, it's not necessary. I thought this is clearer shows the conditions which cause the problem. If it's not obvious, then I'll rather remove this.

> Btw., how would it behave if something was added to that non existent archive? Perhaps this case deserves a separate test.

In this case all right: it just doesn't create an _empty_ archives. 
If an archive is not exist and we add some files then new zip file will be created. 
If an archive already exists and we don't add any files, but use OVERWRITE then it remove the file without warning. I found this is weird, but it works so.
But you right, I'll check the test suite of the extension for coverage this cases.